packages/starknet-snap/src/rpcs/executeTxn.ts

```diff
@@ -0,0 +1,248 @@
+import {
+  AddressStruct,
+  BaseRequestStruct,
+  AccountRpcController,
+  AuthorizableStruct,
+  confirmDialog,
+  InvocationStruct,
+  UniversalDetailsStruct,
+} from '../utils';
+import {
+  ACCOUNT_CLASS_HASH,
+  CAIRO_VERSION,
+  TRANSACTION_VERSION,
+} from '../utils/constants';
+import { addDialogTxt, getTxnSnapTxt } from '../utils/snapUtils';
+import {
+  createAccount,
+  executeTxn as executeTxnUtil,
+  getAccContractAddressAndCallData,
+  getEstimatedFees,
+  isAccountDeployed,
+} from '../utils/starknetUtils';
+
+export const ExecuteTxnRequestStruct = refine(
+  assign(
+    object({
+      address: AddressStruct,
+      invocations: array(InvocationStruct),
+      details: UniversalDetailsStruct,
+      abis: optional(any()),
+    }),
+    AuthorizableStruct,
+    BaseRequestStruct,
+  ),
+  'ExecuteTxnRequestStruct',
+  (value) => {
+    if (value.invocations.length === 0) {
+      return 'Invocations cannot be empty';
+    }
+    for (const invocation of value.invocations) {
+      if (invocation.type !== TransactionType.INVOKE) {
+        return `Invocations should be of type ${TransactionType.INVOKE} received ${invocation.type}`;
+      }
+    }
+    return true;
+  },
+);
+
+export const ExecuteTxnResponseStruct = object({
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  transaction_hash: string(),
+});
+
+export type ExecuteTxnParams = Infer<typeof ExecuteTxnRequestStruct> & Json;
+
+export type ExecuteTxnResponse = Infer<typeof ExecuteTxnResponseStruct>;
+
+const getInvokeCalls = (invocations: Invocations): Call[] => {
+  return invocations
+    .filter((invocation) => invocation.type === TransactionType.INVOKE)
+    .map((invocation: any) => invocation.payload as Call);
+};
+
+/**
+ * The RPC handler to execute a transaction.
+ */
+export class ExecuteTxnRpc extends AccountRpcController<
+  ExecuteTxnParams,
+  ExecuteTxnResponse
+> {
+  protected requestStruct = ExecuteTxnRequestStruct;
+
+  protected responseStruct = ExecuteTxnResponseStruct;
+
+  /**
+   * Execute the transaction request handler.
+   *
+   * @param params - The parameters of the request.
+   * @param params.address - The address of the signer.
+   * @param params.invocations - The invocations to execute (only invocations of type TransactionType.INVOKE)
+   * @param params.abis - The abis associated to invocations.
+   * @param params.details - The detail associated to the call.
+   * @returns The InvokeFunctionResponse as an `ExecuteTxnResponse`.
+   */
+  async execute(params: ExecuteTxnParams): Promise<ExecuteTxnResponse> {
+    return super.execute(params);
+  }
+
+  protected async handleRequest(
+    params: ExecuteTxnParams,
+  ): Promise<ExecuteTxnResponse> {
+    const { address, invocations, abis, details, enableAuthorize } = params;
+
+    const calls = getInvokeCalls(invocations as Invocations);
+
+    const accountDeployed = await isAccountDeployed(this.network, address);
```

Comment:
it seem the getEstimatedFees already tell if you need to account deploy, it seem duplicated